lib/ansible/modules/cloud/vmware/vmware_guest_network.py

```diff
@@ -461,9 +479,31 @@ def get_network_config_spec(self, vm_obj, network_list):
                             if 'connected' in network and nic_device.connectable.connected != network['connected']:
                                 nic_device.connectable.connected = network['connected']
                                 self.change_detected = True
-                            if 'name' in network and nic_device.deviceInfo.summary != network['name']:
-                                nic_device.deviceInfo.summary = network['name']
-                                self.change_detected = True
+                            if 'name' in network:
+                                network_object = self.find_network_by_name(network_name=network['name'])[0]
+                                if network_object and hasattr(network_object, 'portKeys'):
+                                    if network_object.config.distributedVirtualSwitch.uuid != nic_spec.device.backing.port.switchUuid:
+                                        # DistributedVirtualPortGroup
+                                        nic_spec.device.backing = vim.vm.device.VirtualEthernetCard.DistributedVirtualPortBackingInfo()
+                                        nic_spec.device.backing.port = vim.dvs.PortConnection()
+                                        nic_spec.device.backing.port.switchUuid = network_object.config.distributedVirtualSwitch.uuid
+                                        nic_spec.device.backing.port.portgroupKey = network_object.key
+                                        self.change_detected = True
+                                elif network_object and isinstance(network_object, vim.OpaqueNetwork):
```

Comment:
This also gives an error.
The cause is that the `opaqueNetworkId` key does not exist when the portgroup is assigned to the vNIC.
The following conditions are necessary.

```suggestion
                                elif network_object and isinstance(network_object, vim.OpaqueNetwork) and hasattr(nic_spec.device.backing, 'opaqueNetworkId'):
```

tested on vCenter/ESXi 6.7 and NSX-T 2.5